test/e2e/gameserverallocation_test.go

```diff
@@ -589,6 +591,10 @@ func TestCounterAndListGameServerAllocation(t *testing.T) {
 			}
 			require.NoError(t, err)
 			assert.Equal(t, string(testCase.wantAllocated), string(gsa.Status.State))
+			if gsa.Status.State == allocated {
```

Comment:
I thought this test is for counters and lists selector, so I check that the status of counters and lists does not change before/after request.
Since GSA may returns UnAllocated state, check the status only if GSA state is Allocated.

Reply:
This makes sense to me!